dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/blas/Matrix.java

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (C) 2015 Seoul National University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.dolphin.dnn.blas;
+
+/**
+ * Matrix interface.
+ */
+public interface Matrix {
+
+  /**
+   * Returns the number of rows.
+   */
+  int getRows();
+
+  /**
+   * Returns the number of columns.
+   */
+  int getColumns();
+
+  /**
+   * Returns a element specified by the given index (linear indexing).
+   */
+  float get(final int i);
+
+  /**
+   * Returns elements specified by the linear indices.
+   */
+  Matrix get(final int[] indices);
+
+  /**
+   * Returns a element specified by the row and column indices.
+   */
+  float get(final int rowIndex, final int columnIndex);
+
+  /**
+   * Sets a matrix element (linear indexing).
+   */
+  Matrix put(final int i, final float v);
+
+  /**
+   * Sets a matrix element.
+   */
+  Matrix put(final int rowIndex, final int columnIndex, final float v);
+
+  /**
+   * Sets a column with the given column vector.
+   */
+  void putColumn(final int i, Matrix v);
+
+  /**
+   * Sets a row with the given row vector.
+   */
+  void putRow(final int i, Matrix v);
+
+  /**
+   * Returns a copy of a column.
+   */
+  Matrix getColumn(final int i);
+
+  /**
+   * Returns a copy of a row.
+   */
+  Matrix getRow(final int i);
+
+  /**
+   * Returns total number of elements.
+   */
+  int getLength();
+
+  /**
+   * Checks whether the matrix is a column vector.
+   */
+  boolean isColumnVector();
+
+  /**
+   * Checks whether the matrix is a row vector.
+   */
+  boolean isRowVector();
+
+  /**
+   * Sets all elements to the specified value.
+   */
+  Matrix fill(final float v);
+
+  /**
+   * Reshapes the matrix.
+   * The number of elements must not change.
+   */
+  Matrix reshape(final int newRows, final int newColumns);
+
+  /**
+   * Returns a string representation of this matrix.
+   */
+  String toString();
+
+  /**
+   * Converts the matrix to a one-dimensional array of {@code float}s.
+   */
+  float[] toFloatArray();
+
+  /**
+   * Copies {@code Matrix} {@code m} to this.
+   * @param m a source matrix
+   * @return a source matrix {@code m}
```

Comment:
I followed the interface of jblas's one. But, returning `this` seems better since many other methods return `this`. I'll change this.